includes/express-checkout/class-wc-payments-express-checkout-button-helper.php

```diff
@@ -394,6 +394,13 @@ public function should_show_express_checkout_button() {
 
 		// Order total doesn't matter for Pay for Order page. Thus, this page should always display payment buttons.
 		if ( $this->is_pay_for_order_page() ) {
+
+			// ECE doesn't work on the Pay For Order Page.
+			$order = wc_get_order( absint( get_query_var( 'order-pay' ) ) );
+			if ( $order && class_exists( 'WC_Deposits_Order_Manager' ) && WC_Deposits_Order_Manager::is_follow_up_order( $order ) ) {
+				return false;
+			}
```

Comment:
>Is this specifically that ECE doesn't work for WC Deposits on the Pay for Order (PfO) page, or in general? 

I thought it only failed for WC Deposits but it failed in general **for me**. My bad. I've confirmed the issue happened because the Cart Testing protection service was enabled. It'll be fixed on https://github.com/Automattic/woocommerce-payments/pull/9089